common/beerocks/bwl/dwpal/mon_wlan_hal_dwpal.cpp

```diff
@@ -139,10 +139,309 @@ static bool dwpal_get_channel_scan_freq(const std::vector<unsigned int> &channel
     return true;
 };
 
+static bool read_nl_data_from_msg(struct nlattr **bss, struct nl_msg *msg)
+{
+    struct genlmsghdr *gnlh = (genlmsghdr *)nlmsg_data(nlmsg_hdr(msg));
+    struct nlattr *tb[NL80211_ATTR_MAX + 1];
+    static struct nla_policy bss_policy[NL80211_BSS_MAX + 1];
+
+    if (bss == nullptr || msg == nullptr) {
+        LOG(ERROR) << "invalid input bss=" << bss << ", msg=" << msg;
+        return false;
+    }
+
+    bss_policy[NL80211_BSS_BSSID]                = {};
+    bss_policy[NL80211_BSS_FREQUENCY].type       = NLA_U32;
+    bss_policy[NL80211_BSS_TSF].type             = NLA_U64;
+    bss_policy[NL80211_BSS_BEACON_INTERVAL].type = NLA_U16;
+    bss_policy[NL80211_BSS_CAPABILITY].type      = NLA_U16;
+    bss_policy[NL80211_BSS_INFORMATION_ELEMENTS] = {};
+    bss_policy[NL80211_BSS_SIGNAL_MBM].type      = NLA_U32;
+    bss_policy[NL80211_BSS_SIGNAL_UNSPEC].type   = NLA_U8;
+    bss_policy[NL80211_BSS_STATUS].type          = NLA_U32;
+    bss_policy[NL80211_BSS_SEEN_MS_AGO].type     = NLA_U32;
+    bss_policy[NL80211_BSS_BEACON_IES]           = {};
+
+    nla_parse(tb, NL80211_ATTR_MAX, genlmsg_attrdata(gnlh, 0), genlmsg_attrlen(gnlh, 0), NULL);
+
+    if (!tb[NL80211_ATTR_BSS]) {
+        LOG(ERROR) << "tb[NL80211_ATTR_BSS] == NULL";
```

Comment:
This log message can be quite difficult to understand for someone that didn't write this code.
A better message would be: "XXX netlink message is missing the BSS attribute".